care/facility/models/patient.py

```diff
@@ -0,0 +1,161 @@
+
+    past_travel = models.BooleanField(
+        default=False, verbose_name="Travelled to Any Foreign Countries in the last 28 Days",
+    )
+    countries_travelled = models.TextField(default="", blank=True, verbose_name="Countries Patient has Travelled to")
+    date_of_return = models.DateTimeField(
+        blank=True, null=True, verbose_name="Return Date from the Last Country if Travelled"
+    )
+
+    present_health = models.TextField(default="", blank=True, verbose_name="Patient's Current Health Details")
+    ongoing_medication = models.TextField(default="", blank=True, verbose_name="Already pescribed medication if any")
+    has_SARI = models.BooleanField(default=False, verbose_name="Does the Patient Suffer from SARI")
+
+    local_body = models.ForeignKey(LocalBody, on_delete=models.SET_NULL, null=True, blank=True)
+    district = models.ForeignKey(District, on_delete=models.SET_NULL, null=True, blank=True)
+    state = models.ForeignKey(State, on_delete=models.SET_NULL, null=True, blank=True)
+
+    disease_status = models.IntegerField(
+        choices=DISEASE_STATUS_CHOICES, default=1, blank=True, verbose_name="Disease Status"
+    )
+
+    number_of_aged_dependents = models.IntegerField(
+        default=0, verbose_name="Number of people aged above 60 living with the patient", blank=True
+    )
+    number_of_chronic_diseased_dependents = models.IntegerField(
+        default=0, verbose_name="Number of people who have chronic diseases living with the patient", blank=True
+    )
+
+    created_by = models.ForeignKey(User, on_delete=models.SET_NULL, null=True)
+    is_active = models.BooleanField(
+        default=True, help_text="Not active when discharged, or removed from the watchlist",
+    )
+
+    history = HistoricalRecords()
+
+    objects = SoftDeleteManager()
+
+    def __str__(self):
+        return "{} - {} - {}".format(self.name, self.age, self.get_gender_display())
+
+    @staticmethod
+    def has_write_permission(request):
+        return request.user.is_superuser or request.user.user_type >= User.TYPE_VALUE_MAP["Staff"]
+
+    @staticmethod
+    def has_read_permission(request):
+        return True
+
+    def has_object_read_permission(self, request):
+        return (
+            request.user.is_superuser
+            or request.user == self.created_by
+            or (self.facility and request.user == self.facility.created_by)
+            or (
+                request.user.user_type >= User.TYPE_VALUE_MAP["DistrictLabAdmin"]
+                and request.user.district == self.district
+            )
+        )
+
+    def has_object_destroy_permission(self, request):
+        """Currently refers only to delete action"""
+        return request.user.is_superuser
+
+    def has_object_update_permission(self, request):
+        return (
+            request.user.is_superuser
+            or request.user == self.created_by
+            or (self.facility and request.user == self.facility.created_by)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Instance of 'ForeignKey' has no 'created_by' member](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5331745)